src/Asm89/Stack/CorsService.php

```diff
@@ -74,16 +74,18 @@ public function isPreflightRequest(Request $request)
 
     public function addActualRequestHeaders(Response $response, Request $request)
     {
-        if (!$this->checkOrigin($request)) {
-            return $response;
-        }
-
-        $response->headers->set('Access-Control-Allow-Origin', $request->headers->get('Origin'));
+        if ($this->options['supportsCredentials'] || !empty($this->options['allowedOriginsPatterns'])) {
+            if ($this->checkOrigin($request)) {
+                $response->headers->set('Access-Control-Allow-Origin', $request->headers->get('Origin'));
+            }
 
-        if (!$response->headers->has('Vary')) {
-            $response->headers->set('Vary', 'Origin');
+            if (!$response->headers->has('Vary')) {
+                $response->headers->set('Vary', 'Origin');
+            } else {
+                $response->headers->set('Vary', $response->headers->get('Vary') . ', Origin');
+            }
         } else {
-            $response->headers->set('Vary', $response->headers->get('Vary') . ', Origin');
+            $response->headers->set('Access-Control-Allow-Origin', implode(', ', $this->options['allowedOrigins']));
```

Comment:
This could be empty.

Reply:
You are right, so what would expected behavior be then? If both the patterns (case above) and the origins array is empty, no Origin will be considered valid. So don't add the header?

Reply:
Also, I missed this, but `Access-Control-Allow-Origin` only supports a single value. It can either be `*` or a single origin.

If `allowedOrigins` and `allowedOriginsPatterns` are empty, this header should **not** be added. 
if `allowedOriginsPatterns`  is empty, then we know there is a __possability__ of a static value, but the header can only be static *if* there is either a single value in `allowedOrigins` and that value is not `*` OR the value is `*` and `supportsCredentials` is `false`

Reply:
I've updated it:
1. Wildcard + no credentials -> just '*', no Vary
2. Just 1 origin allowed, no wildcard/patterns -> Just the static value, no Vary
3. Always add Vary + check if Origin is allowed. Add origin of allowed, skip otherwise.